src/ZfcUser/Factory/Controller/UserControllerFactory.php

```diff
@@ -25,14 +31,32 @@ class UserControllerFactory implements FactoryInterface
      */
     public function createService(ServiceLocatorInterface $controllerManager)
     {
-        /* @var ControllerManager $controllerManager*/
+        /* @var ControllerManager $controllerManager */
         $serviceManager = $controllerManager->getServiceLocator();
 
         /* @var RedirectCallback $redirectCallback */
         $redirectCallback = $serviceManager->get('zfcuser_redirect_callback');
 
+        /** @var User $userService */
+        $userService = $serviceManager->get('zfcuser_user_service');
+
+        /** @var Login $loginForm */
+        $loginForm = $serviceManager->get('zfcuser_login_form');
+
+        /** @var Register $registerForm */
+        $registerForm = $serviceManager->get('zfcuser_register_form');
+
+        /** @var ChangePassword $changePasswordForm */
+        $changePasswordForm = $serviceManager->get('zfcuser_change_password_form');
+
+        /** @var ChangeEmail $changeEmailForm */
+        $changeEmailForm = $serviceManager->get('zfcuser_change_email_form');
+
+        /** @var ModuleOptions $options */
+        $options = $serviceManager->get('zfcuser_module_options');
+
         /* @var UserController $controller */
-        $controller = new UserController($redirectCallback);
+        $controller = new UserController($userService, $loginForm, $registerForm, $changePasswordForm, $changeEmailForm, $options, $redirectCallback);
```

Comment:
This is true, but the best way to remove the ServiceManager-  and ServiceLocatorAwareInterface is to inject the dependencies via constructor.

The solution that will support BC would require the use of getters that throw exceptions runtime when a class's dependencies aren't set yet.